Add "--verifier clone" testing to many fusion/e2e tests

pcf000 opened the pull request:
This is most of issue #683, I think.  Two tosa-to-rock tests aren't included because they don't work yet, and two mixr-gemm tests have different answers when partitioning and that's weird.

sjw36 commented on mlir/test/fusion/e2e/mixr-gemm/mixr-gemm-add.e2e.mlir:
```diff
@@ -1,11 +1,20 @@
   // CHECK-NEXT: [5,     5,     5]
+
+  // CLONE: [1 1 1]
+  // CLONE-NEXT: Unranked Memref base
```
Let's use '-rand 1 -rand_type float' for all these so we are sure it is working. And you can just check for '[1 1 1]' then.

pcf000 replied:
Done, for all but tosa-to-rock-rsqrt.e2e.mlir.  For that one, the new random options caused all the result values to be NaN, I think because intermediate values were negative.  Left it with "-rand none".

sjw36 reviewed (changes requested):
See comment.

krzysz00 commented:
For the sqrt, it might be a good idea to go with `-rand fixed`

sjw36 reviewed (approved):
Looks good.

pcf000 committed:
Restore the clone-verification command for resnet50_blk.e2e.mlir.
Add a clone-verification case to six of the tosa fusion/e2e tests.
Add clone-verifier cases to the mixr-gemm tests, and an explanatory string to an assertion.
